av_extend_guts Test Failure in 0.26

I'm running into an issue upgrading this POD from 0.25 to 0.26, and it appears this issue has been encountered by others[*](#star).

My install log is as follows:
```
cpan[9]> upgrade Cache::Memcached::Fast                                                                               

Package namespace         installed    latest  in CPAN file
Cache::Memcached::Fast         0.25      0.26  RAZ/Cache-Memcached-Fast-0.26.tar.gz
Running install for module 'Cache::Memcached::Fast'
Checksum for /root/.cpan/sources/authors/id/R/RA/RAZ/Cache-Memcached-Fast-0.26.tar.gz ok
Configuring R/RA/RAZ/Cache-Memcached-Fast-0.26.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for Cache::Memcached::Fast
Writing MYMETA.yml and MYMETA.json
RAZ/Cache-Memcached-Fast-0.26.tar.gz
/usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK
Running make for R/RA/RAZ/Cache-Memcached-Fast-0.26.tar.gz
cp lib/Cache/Memcached/Fast.pm blib/lib/Cache/Memcached/Fast.pm
cd src && make LIBPERL_A="libperl.a" LINKTYPE="dynamic" OPTIMIZE="-O2 -g" PREFIX="/usr/local" PASTHRU_DEFINE=' ' PASTHRU_INC=' '
make[1]: Entering directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
"/usr/bin/perl" genparser.pl reply.kw parse_keyword.c parse_keyword.h
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H parse_keyword.c
"/usr/bin/perl" gencrc32.pl compute_crc32.c compute_crc32.h
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H compute_crc32.c
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H array.c
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H client.c
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H connect.c
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H dispatch_key.c
x86_64-linux-gnu-gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"  -DHAVE_POLL_H socket_posix.c
ar cr libclient.a parse_keyword.o compute_crc32.o array.o client.o connect.o dispatch_key.o socket_posix.o
: libclient.a
make[1]: Leaving directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
make[1]: Entering directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
Running Mkbootstrap for Fast ()
chmod 644 "Fast.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Fast.bs blib/arch/auto/Cache/Memcached/Fast/Fast.bs 644
"/usr/bin/perl" "/usr/share/perl/5.30/ExtUtils/xsubpp"  -typemap '/usr/share/perl/5.30/ExtUtils/typemap' -typemap '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/typemap'  Fast.xs > Fast.xsc
mv Fast.xsc Fast.c
x86_64-linux-gnu-gcc -c   -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"0.26\" -DXS_VERSION=\"0.26\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.30/CORE"   Fast.c
rm -f blib/arch/auto/Cache/Memcached/Fast/Fast.so
x86_64-linux-gnu-gcc  -shared -L/usr/local/lib -fstack-protector-strong  Fast.o  -o blib/arch/auto/Cache/Memcached/Fast/Fast.so src/libclient.a \
\

chmod 755 blib/arch/auto/Cache/Memcached/Fast/Fast.so
Manifying 1 pod document
RAZ/Cache-Memcached-Fast-0.26.tar.gz
/usr/bin/make -- OK
Running make test for RAZ/Cache-Memcached-Fast-0.26.tar.gz
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Fast.bs blib/arch/auto/Cache/Memcached/Fast/Fast.bs 644
make[1]: Entering directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
make[1]: Entering directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
No tests defined for Cache::Memcached::Fast::libclient extension.
make[1]: Leaving directory '/root/.cpan/build/Cache-Memcached-Fast-0.26-2/src'
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-load.t ......... 1/1 # Testing Cache::Memcached::Fast 0.26, Perl 5.030000, /usr/bin/perl
t/00-load.t ......... ok   
t/01-connect.t ...... # Connected to 2 memcached servers, lowest version 1.5.22
t/01-connect.t ...... ok   
t/02-isa.t .......... ok   
t/big_value.t ....... ok     
t/commands.t ........ 69/126 panic: av_extend_guts() negative count (-2) at t/commands.t line 175.
# Looks like your test exited with 11 just after 70.
t/commands.t ........ Dubious, test returned 11 (wstat 2816, 0xb00)
Failed 56/126 subtests 
t/errors.t .......... ok   
t/hash_namespace.t .. ok   
t/key_ref.t ......... ok   
t/magic.t ........... ok   
t/malformed.t ....... ok     
t/namespace.t ....... ok   
t/noreply.t ......... ok   
t/nowait.t .......... ok   
t/pod-coverage.t .... ok
t/pod.t ............. ok
t/serialize.t ....... ok
t/threads.t ......... ok
t/utf8-downgrade.t .. ok
t/utf8.t ............ ok

Test Summary Report
-------------------
t/commands.t      (Wstat: 2816 Tests: 70 Failed: 0)
Non-zero exit status: 11
Parse errors: Bad plan.  You planned 126 tests but ran 70.
Files=19, Tests=179, 15 wallclock secs ( 0.08 usr  0.02 sys +  2.02 cusr  0.29 csys =  2.41 CPU)
Result: FAIL
Failed 1/19 test programs. 0/179 subtests failed.
make: *** [Makefile:1043: test_dynamic] Error 255
RAZ/Cache-Memcached-Fast-0.26.tar.gz
/usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
reports RAZ/Cache-Memcached-Fast-0.26.tar.gz
Failed during this command:
RAZ/Cache-Memcached-Fast-0.26.tar.gz         : make_test NO
```

System Info:
- Ubuntu 20.04.1 LTS
- Perl v5.30.0
- Linux 5.4.0-1032-aws #33-Ubuntu SMP Wed Dec 9 17:13:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

<span id="star">* An identical issue was opened in the (soon-to-be-decommissioned) RT at [cpan.org](https://rt.cpan.org/Ticket/Display.html?id=132572), but doesn't seem to have been recreated here.</span>

--- a/Fast.xs
+++ b/Fast.xs
@@ -1043,7 +1043,8 @@ gat_multi(memd, ...)
         value_res.memd = memd;
         value_res.vals = (SV *) newAV();
         sv_2mortal(value_res.vals);
-        av_extend((AV *) value_res.vals, key_count - 1);
+        if (key_count > 1)
+          av_extend((AV *) value_res.vals, key_count - 1);
         client_reset(memd->c, &object, 0);
         sv = ST(1);
         SvGETMAGIC(sv);

--- a/lib/Cache/Memcached/Fast.pm
+++ b/lib/Cache/Memcached/Fast.pm
@@ -1234,7 +1234,7 @@ B<gat> command first appeared in B<memcached> 1.5.3.
 
 =item C<gats>
 
-  $memd->gets($expiration_time, $key);
+  $memd->gats($expiration_time, $key);
 
 Update the expiration time and Retrieve the value and its CAS for a I<$key>.
 

